public/app/core/utils/CorsSharedWorker.ts

```diff
@@ -0,0 +1,32 @@
+// Almost identical to CorsWorker.ts. Main difference being it allows loading a SharedWorker if browser supports it
+
+export function sharedWorkersSupported() {
+  return typeof window.SharedWorker !== 'undefined';
+}
+
+/**
+ * Creating CorsSharedWorker should be called only if sharedWorkersSupported() is truthy
+ */
+export class CorsSharedWorker {
+  constructor(url: URL, options?: WorkerOptions) {
+    // by default, worker inherits HTML document's location and pathname which leads to wrong public path value
+    // the CorsWorkerPlugin will override it with the value based on the initial worker chunk, ie.
```

Comment:
```suggestion
    // the CorsSharedWorker will override it with the value based on the initial worker chunk, ie.
```

Reply:
This one is actually correct. There's a single plugin for Web and SharedWorkers